packages/core/src/components/BaseInput/__tests__/BaseInput.jest.tsx

```diff
@@ -0,0 +1,109 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import userEvent from "@testing-library/user-event";
+import BaseInput from "../BaseInput";
+import { BaseInputProps } from "../BaseInput.types";
+
+function renderBaseInput(props?: Partial<BaseInputProps>) {
+  return render(<BaseInput aria-label="base-input" {...props} />);
+}
+
+describe("BaseInput", () => {
+  it("should render correctly", () => {
+    const { getByLabelText } = renderBaseInput();
+    expect(getByLabelText("base-input")).toBeInTheDocument();
+  });
+
+  describe("with declared props", () => {
+    it("should apply the size class", () => {
+      const { getByLabelText } = renderBaseInput({ size: "large" });
+      expect(getByLabelText("base-input").parentNode).toHaveClass("large");
```

Comment:
I think you can use `container` or something similar to get the outer element

Reply:
container gives where the BaseInput is rendered, so I need to query the there the `firstChild` so it is pretty much the same..
I'll leave it the way it is I think